clients/drcachesim/analyzer_multi.cpp

```diff
@@ -106,26 +126,231 @@ analyzer_multi_t::create_external_tool(const std::string &tool_name)
+    } else if (simulator_type == BASIC_COUNTS) {
+        return basic_counts_tool_create(op_verbose.get_value());
+    } else if (simulator_type == OPCODE_MIX) {
+        std::string module_file_path = get_module_file_path();
+        if (module_file_path.empty() && op_indir.get_value().empty() &&
+            op_infile.get_value().empty() && !op_instr_encodings.get_value()) {
+            ERRMSG("Usage error: the opcode_mix tool requires offline traces, or "
+                   "-instr_encodings for online traces.\n");
+            return nullptr;
+        }
+        return opcode_mix_tool_create(module_file_path, op_verbose.get_value(),
+                                      op_alt_module_dir.get_value());
+    } else if (simulator_type == SYSCALL_MIX) {
+        return syscall_mix_tool_create(op_verbose.get_value());
+    } else if (simulator_type == VIEW) {
+        std::string module_file_path = get_module_file_path();
+        // The module file is optional so we don't check for emptiness.
+        return view_tool_create(module_file_path, op_skip_refs.get_value(),
+                                op_sim_refs.get_value(), op_view_syntax.get_value(),
+                                op_verbose.get_value(), op_alt_module_dir.get_value());
+    } else if (simulator_type == FUNC_VIEW) {
+        std::string funclist_file_path = get_aux_file_path(
+            op_funclist_file.get_value(), DRMEMTRACE_FUNCTION_LIST_FILENAME);
+        if (funclist_file_path.empty()) {
+            ERRMSG("Usage error: the func_view tool requires offline traces.\n");
+            return nullptr;
+        }
+        return func_view_tool_create(funclist_file_path, op_show_func_trace.get_value(),
+                                     op_verbose.get_value());
+    } else if (simulator_type == INVARIANT_CHECKER) {
+        return create_invariant_checker();
+    } else if (simulator_type == SCHEDULE_STATS) {
+        return schedule_stats_tool_create(op_schedule_stats_print_every.get_value(),
+                                          op_verbose.get_value());
+    } else {
+        auto tool = create_external_tool(simulator_type);
+        if (tool == nullptr) {
+            ERRMSG("Usage error: unsupported analyzer type \"%s\". "
+                   "Please choose " CPU_CACHE ", " MISS_ANALYZER ", " TLB ", " HISTOGRAM
+                   ", " REUSE_DIST ", " BASIC_COUNTS ", " OPCODE_MIX ", " SYSCALL_MIX
+                   ", " VIEW ", " FUNC_VIEW ", or some external analyzer.\n",
+                   simulator_type.c_str());
+        }
+        return tool;
+    }
+}
+
+/******************************************************************************
+ * Specializations for analyzer_multi_tmpl_t<trace_entry_t, record_reader_t>, aka
+ * record_analyzer_multi_t.
+ */
+
+template <>
+std::unique_ptr<record_reader_t>
+record_analyzer_multi_t::create_ipc_reader(const char *name, int verbose)
+{
+    // Not supported;
+    return std::unique_ptr<record_reader_t>();
```

Comment:
This does not create a record_reader_t: it creates an empty unique_ptr.  So it is already returning the only equivalent of nullptr that it can return.

Reply:
error_string_ was added to all of these.